### ISSUE TITLE
[Question] static_operator with infix

### ISSUE THREAD

**Issue by NikKovIos** [realm/SwiftLint#5656]
<img width="1340" alt="image" src="https://github.com/realm/SwiftLint/assets/16647690/1e3c31cb-2457-4e0c-a03b-3ae1f466e906">
<img width="1349" alt="image" src="https://github.com/realm/SwiftLint/assets/16647690/fbe3effb-fc73-46a7-85c7-663bb0d77ae2">

Can't figure out how to declare infix operator without warning. Help please.

**Comment by SimplyDanny** [comment]
This should work:

```swift
infix operator +?

extension Int {
  static func +? (a: Self?, b: Self) -> Self? {
    if let a { a + b } else { nil }
  }
}
```

The idea is that operators should be static and associated with the type. But the operator declaration needs to go to file scope. It cannot be part of the extension.

**Comment by NikKovIos** [comment]
Work!
